[BOLT11] Feature Bits MUST be required (#747)
lightningnetwork/lightning-rfc: 11-payment-encoding.md

### Requirements

A writer:

>   - MUST include exactly one `p`, `9` (with at least the `payment_secret` support bit set), and `s` fields.

Reviewer: I believe this would be too specific.
What's probably confusing the requirement to include an `s` field without any mention of the feature bits.
Here is what I propose:

```suggestion
  - MUST include exactly one `p` field.
  - If the `payment_secret` feature is set, MUST include exactly one `s` field.
```

Reviewer: Note that the requirements around features are in the `Feature Bits` section, so we could even remove that additional requirement which ends up being duplicated.

```suggestion
  - MUST include exactly one `p` field.
```

  - if `9` contains non-zero bits:
    - SHOULD use the minimum `data_length` possible.
  - otherwise:

>     - MUST omit the `9` and `s` field altogether and follow the old BOLT11 protocol.

Reviewer: I don't think it's a good idea to mention an "old BOLT11 protocol".
The spec should only reflect the latest protocol (which is backwards-compatible).
I would remove this.

  - if `9` contains non-zero bits:
    - SHOULD use the minimum `data_length` possible.
  - otherwise:

>     - MUST omit the `9` and `s` field altogether.

Reviewer: I think you can remove that change. A requirement a bit further down already specifies that: "MUST set an s field if and only if the payment_secret feature is set."
